### PR TITLE
DOC: Correct misuse of term high-cardinality in docs.

### PR DESCRIPTION
**Pull request by ebrassell**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
cardinality seems like a weird term to use here.  can you suggest a clearer one?

**Comment by jorisvandenbossche** [comment]
It's indeed a very unclear term if you don't know it, but it's also commonly used in the machine learning community for this purpose (so it's certainly a correct term, at least with the fix here). 
Since it is explain afterwards, I am fine with keeping it. But maybe switching it around (first the more easy explanation, then the technical term might be better)? Eg `This is especially true for text data with only few unique values (low-cardinality data).`

**Comment by ebrassell** [comment]
I agree that _cardinality_ is the correct term here and certainly expect the target audience to be familiar with its proper usage, but I've reworded it as suggested for the benefit of those that might not be familiar with the term.

**Review by WillAyd — APPROVED** [review]
lgtm

File: doc/source/user_guide/scale.rst [hunk 1]
```diff
@@ -93,9 +93,9 @@ Use efficient datatypes
 -----------------------
 
 The default pandas data types are not the most memory efficient. This is
-especially true for high-cardinality text data (columns with relatively few
-unique values). By using more efficient data types you can store larger datasets
-in memory.
+especially true for text data columns with relatively few unique values (commonly
+referred to as "low-cardinality" data). By using more efficient data types you
```
Comment (gfyoung) [inline doc/source/user_guide/scale.rst]:
```suggestion
referred to as "low-cardinality" data). By using more efficient data types, you
```

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jreback** [comment]
@gfyoung can you commit your suggestion and merge.

**Comment by gfyoung** [comment]
@jreback : This PR didn't allow for maintainers to make changes.

**Comment by gfyoung** [comment]
I will merge this and follow-up with the grammar fix.

**Comment by gfyoung** [comment]
Thanks @ebrassell !

File: doc/source/user_guide/scale.rst (hunk 1 above)
Comment (gfyoung) [reply doc/source/user_guide/scale.rst]: Fixed in https://github.com/pandas-dev/pandas/pull/29848

**Pull request merged**